### PR TITLE
insteonplm: add support for smoke bridge (this time for real)

### PR DESCRIPTION
**Pull request by berndpfrommer**
Apologies, I messed up on PR #5026: it did not contain the smoke bridge code.
This PR should fix it.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/device/MessageHandler.java [hunk 1]
```diff
@@ -971,16 +971,20 @@ public int transform(int raw) {
             return (raw);
         }
 
-        private int extractValue(Msg msg) throws FieldException {
+        private int extractValue(Msg msg, int group) throws FieldException {
             String lowByte = getStringParameter("low_byte", "");
-            if (lowByte == "") {
+            if (!lowByte.equals("")) {
```
Comment (henwid) [inline bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/device/MessageHandler.java]: This code change has inverted the test and now (improperly) fails consistently. Insteon Smoke Bridge not working and Insteon Thermostat no longer working i.e. impossible to retrieve status and config values from thermostat during poll(?)

> `2017-03-24 00:24:36.073 [ERROR] [onplm.internal.device.MessageHandler] - NumberMsgHandler handler misconfigured, missing low_byte!

Comment (berndpfrommer) [inline bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/device/MessageHandler.java]: Dang it, yes that's a bug. If you fix that one line, does the smokebridge work?

Comment (henwid) [inline bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/device/MessageHandler.java]: I wish I could answer that for you! I've put using Eclipse on hold temporarily. If you could make the change and get it into a nightly build I'd be more than happy to let you know.

Comment (berndpfrommer) [inline bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/device/MessageHandler.java]: Here are the jar and .deb files with the fix. I hope the links work.

https://drive.google.com/file/d/0B8OtzwATWQSWbzJzUGNCVldmczA/view?usp=sharing
https://drive.google.com/file/d/0B8OtzwATWQSWcDNWenpOYVRKWms/view?usp=sharing

**Comment by henwid** [comment]
Hi Bernd,Sorry for the delay in getting back to you. I first missed your speedy reply and since then have tried a couple of times to try out the Insteon jar file you provided. Well, things didn't go smoothly. What I thought would just take a few minutes to check out turned into a few long failed sessions! Today I had some time again so I redid everything to document the issue that was slowing things down.Using the Paper UI, I first uninstalled the InsteonPLM binding. As you can see from the messages, this step alone took 23 minutes.2017-04-22 18:48:05.060 [ERROR] [ding.insteonplm.internal.driver.Port] - got interrupted exception in write thread 2017-04-22 18:51:29.728 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 77 configured, 44 polling, msgs received: 219 2017-04-22 19:01:29.749 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 77 configured, 44 polling, msgs received: 0 2017-04-22 19:09:41.336 [ERROR] [ding.insteonplm.internal.driver.Port] - reader thread exiting! 2017-04-22 19:09:41.491 [WARN ] [nding.insteonplm.InsteonPLMActivator] - Insteon PLM binding has been stopped. 2017-04-22 19:09:41.794 [INFO ] [core.karaf.internal.FeatureInstaller] - Uninstalled 'openhab-binding-insteonplm1' 2017-04-22 19:11:29.750 [INFO ] [b.core.service.AbstractActiveService] - InsteonPLM has been shut downI then did a logout from karaf and it shut down very quickly. I then put the jar file into the addons directory, power cycled the Insteon 2012 hub and starte_debugged things up. Here is the Karaf listing showing your binding running:openhab> la | grep nste227 | Active | 80 | 1.8.3.201701160324 | openHAB Insteon PLM Bindingopenhab>Here are the log entries from the startup. As you can see, the new jar file seems unable to complete a modem database download. The current snapshot struggles a few times, too, but I have not been able to get your jar file to succeed any of the times I've tried.2017-04-22 19:19:58.134 [INFO ] [.dashboard.internal.DashboardService] - Started dashboard at /start 2017-04-22 19:19:58.286 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - dead device timeout set to 3000s 2017-04-22 19:19:58.287 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - config: port_0 -> /hub/192.168.1.111:9761 2017-04-22 19:19:58.402 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - config: service.pid -> org.openhab.insteonplm 2017-04-22 19:19:58.771 [INFO ] [b.core.service.AbstractActiveService] - InsteonPLM has been started 2017-04-22 19:19:58.942 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 0 configured, 0 polling, msgs received: 0 2017-04-22 19:19:59.221 [ERROR] [ding.insteonplm.internal.driver.Port] - reader thread exiting! 2017-04-22 19:20:01.077 [INFO ] [basic.internal.servlet.WebAppServlet] - Started Basic UI at /basicui/app 2017-04-22 19:20:01.287 [INFO ] [assic.internal.servlet.WebAppServlet] - Started Classic UI at /classicui/app 2017-04-22 19:20:01.389 [INFO ] [arthome.ui.paper.internal.PaperUIApp] - Started Paper UI at /paperui 2017-04-22 19:21:59.091 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:23:59.092 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:25:59.094 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:27:59.095 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! So, at that point, I did a logout from Karaf to bring the system down. As you can see, the download continues to try repeatedly (and is continuing to do so as I write this email) preventing the karaf from shutting down.2017-04-22 19:35:18.505 [INFO ] [basic.internal.servlet.WebAppServlet] - Stopped Basic UI ....2017-04-22 19:35:59.099 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:37:59.101 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:39:58.945 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 75 configured, 0 polling, msgs received: 0 2017-04-22 19:39:59.103 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:41:59.104 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:43:59.107 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:45:59.109 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:47:59.111 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:49:58.947 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 75 configured, 0 polling, msgs received: 0 2017-04-22 19:49:59.112 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:51:59.113 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:53:59.115 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:55:59.117 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting!2017-04-22 19:57:59.119 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 19:59:58.949 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 75 configured, 0 polling, msgs received: 0 2017-04-22 19:59:59.120 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:01:59.121 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:03:59.175 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:05:59.235 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:07:59.237 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:09:58.951 [INFO ] [g.insteonplm.InsteonPLMActiveBinding] - devices: 75 configured, 0 polling, msgs received: 0 2017-04-22 20:09:59.239 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:11:59.241 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:13:59.242 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! 2017-04-22 20:15:59.246 [WARN ] [onplm.internal.device.ModemDBBuilder] - modem database download unsuccessful, restarting! At this point, all I can do is close the window running karaf.Attempting restart produces:C:\openHAB2>start_debugLaunching the openHAB runtime...karaf.bat: Ignoring predefined value for KARAF_HOMEfind: :: Invalid argumentThere is a Root instance already running with name main and pid 5068C:\openHAB2>I'm now forced to reinstall or go back to a backup. So, as you can see, what should have been a simple test has turned into a frustrating exercise. I'm giving you the details because you may also be able to quickly solve this. It's making doing anything with the InsteonPLM installed a problem. So, long story short, I was not even able to get the jar file running, much less tested.Thanks for your attention. I'm not sure what the next step is but I'm willing to keep trying until these minor issues get solved. Please let me know if you need me to try again. I'll watch that email account more closely, this time, and try to get back to you much more quickly.Thanks again,HenryP.S. WIndows 8.1, 64 bit Java, openHAB SNAPSHOT #889
On April 16, 2017 at 8:20 PM Bernd Pfrommer <notifications@github.com> wrote:@berndpfrommer commented on this pull request.



In bundles/binding/org.openhab.binding.insteonplm/src/main/java/org/openhab/binding/insteonplm/internal/device/MessageHandler.java:
>              String lowByte = getStringParameter("low_byte", "");
-            if (lowByte == "") {
+            if (!lowByte.equals("")) {

Here are the jar and .deb files with the fix. I hope the links work.
https://drive.google.com/file/d/0B8OtzwATWQSWbzJzUGNCVldmczA/view?usp=sharing
https://drive.google.com/file/d/0B8OtzwATWQSWcDNWenpOYVRKWms/view?usp=sharing

**Pull request merged**